File: PokeKotlinApiSample/app/src/main/java/com/example/pokekotlinapisample/MainActivity.java
package com.example.pokekotlinapisample;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.bumptech.glide.Glide;

import me.sargunvohra.lib.pokekotlin.client.PokeApi;
import me.sargunvohra.lib.pokekotlin.client.PokeApiClient;
import me.sargunvohra.lib.pokekotlin.model.Pokemon;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        EditText editText = findViewById(R.id.editTextPokemonId);
        ImageView imageView = findViewById(R.id.imagePokemon);
        TextView textView = findViewById(R.id.txtPokemonIdWithName);

        findViewById(R.id.buttonSearchPokemon).setOnClickListener(v -> {
            new Thread(() -> {
                PokeApi pokeApi = new PokeApiClient();
                try {
                    Pokemon pokemon = pokeApi.getPokemon(Integer.parseInt(editText.getText().toString()));
                    runOnUiThread(() -> {
                        textView.setText(pokemon.getId() + " : " + pokemon.getName());
                        Glide.with(MainActivity.this).load(pokemon.getSprites().component1()).into(imageView);
                    });
                } catch (Throwable throwable) {
                    runOnUiThread(() -> {
                        Toast.makeText(MainActivity.this, "error: " + throwable.getLocalizedMessage(), Toast.LENGTH_SHORT).show();
                    });
                }
            }).start();
        });
    }
}

File: PokeKotlinApiSample2/app/src/main/java/com/example/pokekotlinapisample/MainActivity.kt
package com.example.pokekotlinapisample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.ImageView
import android.widget.TextView
import me.sargunvohra.lib.pokekotlin.client.PokeApi
import me.sargunvohra.lib.pokekotlin.client.PokeApiClient
import com.bumptech.glide.Glide
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val editText = findViewById<EditText>(R.id.editTextPokemonId)
        val imageView = findViewById<ImageView>(R.id.imagePokemon)
        val textView = findViewById<TextView>(R.id.txtPokemonIdWithName)
        findViewById<View>(R.id.buttonSearchPokemon).setOnClickListener {
            Thread {
                val pokeApi: PokeApi = PokeApiClient()
                kotlin.runCatching {
                    pokeApi.getPokemon(editText.text.toString().toInt())
                }.onSuccess { pokemon ->
                    runOnUiThread {
                        textView.text = "${pokemon.id} : ${pokemon.name}"
                        Glide.with(this@MainActivity).load(pokemon.sprites.backDefault)
                            .into(imageView)
                    }
                }.onFailure { throwable ->
                    runOnUiThread {
                        Toast.makeText(
                            this@MainActivity,
                            "error: " + throwable.localizedMessage,
                            Toast.LENGTH_SHORT
                        ).show()
                    }
                }
            }.start()
        }
    }
}